src/models/tools/geometry/jxg-comment.ts

```diff
@@ -129,6 +131,8 @@ export const commentChangeAgent: JXGChangeAgent = {
         }
         board.update();
       }
+      // other properties can be handled generically
+      objectChangeAgent.update(board, change);
```

Comment:
Ah I guess this gives us things like size handling for free, nice

Reply:
@kswenson Looks like this line actually caused a reversion of the undo behavior, still trying to figure out exactly why

Reply:
Ah, looks like the annotation location hasn't had a chance to update yet, so the object change agent causes the relative change to get applied twice. Adding an update after the position is set seems to fix the problem

Reply:
Pushed a proposed change that simplifies things and takes advantage of the fact that `unsuspendUpdate` just does nothing if redrawing is not suspended

Reply:
Adding the call to update() makes sense. I think this code should leave things in the same un/suspend state that it started with. Right now if we started not suspended this code will leave us suspended. How about something like:
```typescript
const wasSuspended = board.isSuspendedUpdate;
if (wasSuspended) board.unsuspendUpdate();
obj.setPosition(JXG.COORDS_BY_USER, position);
board.update();
if (wasSuspended) board.suspendUpdate();
```

Reply:
Yes definitely, good catch!